File: 프로그래머스/2/131127. 할인 행사/할인 행사.py
def solution(want, number, discount):
    answer = 0
    idx = 0
    while(idx <= len(discount)-10):
        arr = discount[idx:idx+10]
        check = False
        for i in range(len(want)):
            if arr.count(want[i]) >= number[i]:
                check = True
            else:
                check = False
                break
        if check: answer+=1
        idx+=1
            
    return answer

File: 프로그래머스/2/131127. 할인 행사/할인 행사.js
function solution(want, number, discount) {
    var answer = 0;
    const arr = {}
    for(let i=0; i<want.length; i++){
        arr[want[i]] = number[i];
    }
    
    for(let i=0; i<discount.length-9; i++){
        let temp = {...arr};
        let discount_slice = discount.slice(i,i+10);
        
        for(let j=0; j<discount_slice.length; j++){
            if(discount_slice[j] in temp){
                temp[discount_slice[j]] -= 1;
            
                if(temp[discount_slice[j]] === 0){
                delete temp[discount_slice[j]];
            }
        }
        }
        if(Object.keys(temp).length === 0){
            answer+=1;
        }
    }
    return answer;
}